src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/inmemory/PetRepositoryInMemoryInterpreter.scala

```diff
@@ -43,8 +43,10 @@ class PetRepositoryInMemoryInterpreter[F[_]: Applicative] extends PetRepositoryA
   def findByStatus(statuses: NonEmptyList[PetStatus]): F[List[Pet]] =
     cache.values.filter(p => statuses.exists(_ == p.status)).toList.pure[F]
 
-  def findByTag(tags: NonEmptyList[String]): F[List[Pet]] =
-    cache.values.filter(p => tags.exists(_ == p.tags)).toList.pure[F]
+  def findByTag(tags: NonEmptyList[String]): F[List[Pet]] = {
+    val tagSet = tags.toNes
+    cache.values.filter(_.tags.exists(tagSet contains_ _)).toList.pure[F]
```

Comment:
I am good with this, makes sense.  Can we add some parens in that.  It is really tough to read.

Other than that, I am good, nice job

Reply:
Thank you very much 😃